### ISSUE TITLE
Monitor not deleted upon removing Ingress

### ISSUE THREAD

**Issue by ejdejesu** [stakater/IngressMonitorController#192]
**Kubernetes Version**: 1.14.1
**Ingress Monitor Controller Version**: 1.0.47
**Provider**: Status Cake

Deleted the ingress with IMC annotation, but the monitor in StatusCake was not deleted.

```
2019/06/26 19:13:00 Warning: KUBERNETES_NAMESPACE is unset, will monitor ingresses in all namespaces.
2019/06/26 19:13:00 Reading YAML Configuration
2019/06/26 19:13:00 Starting Ingress Monitor controller
2019/06/26 19:13:27 Monitor Name: ingress-monitor-test-ingress-statuscake-monitor
2019/06/26 19:13:27 Monitor URL: https://example.com/
2019/06/26 19:13:30 json: cannot unmarshal number 87.16 into Go struct field StatusCakeMonitorMonitor.Uptime of type int
2019/06/26 19:13:31 Monitor Added: 5020592
2019/06/26 19:14:04 json: cannot unmarshal number 87.16 into Go struct field StatusCakeMonitorMonitor.Uptime of type int
2019/06/26 19:14:04 Cannot find monitor with name: ingress-monitor-test-ingress-statuscake-monitor
```

**Comment by usamaahmadkhan** [comment]
@ejdejesu  have you enabled this? https://github.com/stakater/IngressMonitorController/blob/835e24345b8c124dc26e16d60964ff6915efb3ad/deployments/kubernetes/ingressmonitorcontroller.yaml#L192

Also I suggest you upgrade to the latest version.

**Comment by briananstett** [comment]
**Kubernetes Version: 1.13.6**
**Ingress Monitor Controller Version: 1.0.77**
**Provider: Status Cake**

I too am having this issue. The monitor is created in status cake but fails to delete the monitor once the ingress has been deleted.

*Snippet from configmap*
```
data:
  config.yaml: |-
    providers:
    - name: StatusCake
      apiKey: 123456789xxxxx
      apiURL: https://app.statuscake.com/API/
      username: foo
      password: null
      alertContacts: foo_bar
    enableMonitorDeletion: true
    monitorNameTemplate: "{{.IngressName}}-statuscake-monitor"
```
Below are the logs I get when creating the ingress and then promptly removing it. 
```
2019/06/27 13:53:34 Monitor Name: ingress-monitor-test-ingress-statuscake-monitor
2019/06/27 13:53:34 Monitor URL: https://monitor.foo.dev/
time="2019-06-27T13:53:36Z" level=info msg="json: cannot unmarshal number 99.37 into Go struct field StatusCakeMonitorMonitor.Uptime of type int"
time="2019-06-27T13:53:37Z" level=info msg="Monitor Added: 5040053"
time="2019-06-27T13:53:56Z" level=info msg="json: cannot unmarshal number 99.37 into Go struct field StatusCakeMonitorMonitor.Uptime of type int"
2019/06/27 13:53:56 Cannot find monitor with name: ingress-monitor-test-ingress-statuscake-monitor
```

**Comment by usamaahmadkhan** [comment]
@ejdejesu @briananstett `enableMonitorDeletion` annotation is working as expected. This issue is arising for monitors having uptime in decimal percentages. Tests run quickly that is why this use case could not be tested (because uptime is always 100%). this effects all versions. I'll make a PR soon and post an update on this thread. Thanks for reporting this issue

**Comment by usamaahmadkhan** [comment]
fixed in v1.0.82
